packages/ethereum-contracts/contracts/superfluid/FlowNFTBase.sol

```diff
@@ -0,0 +1,389 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity >=0.8.4;
+
+import { UUPSProxiable } from "../upgradability/UUPSProxiable.sol";
+import {
+    IERC165,
+    IERC721,
+    IERC721Metadata
+} from "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { IFlowNFTBase } from "../interfaces/superfluid/IFlowNFTBase.sol";
+import { ISuperfluid } from "../interfaces/superfluid/ISuperfluid.sol";
+import { ISuperToken } from "../interfaces/superfluid/ISuperToken.sol";
+import {
+    IConstantFlowAgreementV1
+} from "../interfaces/agreements/IConstantFlowAgreementV1.sol";
+
+/// @title FlowNFTBase abstract contract
+/// @author Superfluid
+/// @notice The abstract contract to be inherited by the Flow NFTs.
+/// @dev This contract inherits from IFlowNFTBase which inherits from
+/// IERC721Metadata and holds shared storage and functions for the two NFT contracts.
+/// This contract is upgradeable and it inherits from our own ad-hoc UUPSProxiable contract which allows.
+/// NOTE: the storage gap allows us to add an additional 16 storage variables to this contract without breaking child
+/// COFNFT or CIFNFT storage.
+abstract contract FlowNFTBase is UUPSProxiable, IFlowNFTBase {
+    using Strings for uint256;
+
+    string public constant BASE_URI =
+        "https://nft.superfluid.finance/cfa/v1/getmeta";
+
+    /**************************************************************************
+     * Storage variables
+     *************************************************************************/
+    /// NOTE: The storage variables in this contract MUST NOT:
+    /// - change the ordering of the existing variables
+    /// - change any of the variable types
+    /// - rename any of the existing variables
+    /// - remove any of the existing variables
+    /// - add any new variables after _gap
+    /// - add any new variables before _gap and NOT decrement the length of the _gap array
+    /// Go to CFAv1NFTUpgradability.t.sol for the tests and make sure to add new tests for upgrades.
+
+    /// @notice ConstantFlowAgreementV1 contract address
+    /// @dev This is the address of the CFAv1 contract cached so we don't have to
+    /// do an external call for every flow created.
+    // solhint-disable-next-line var-name-mixedcase
+    IConstantFlowAgreementV1 public immutable CONSTANT_FLOW_AGREEMENT_V1;
+
+    ISuperToken public superToken;
+
+    string internal _name;
+    string internal _symbol;
+
+    /// @notice Mapping for token approvals
+    /// @dev tokenID => approved address mapping
+    mapping(uint256 => address) internal _tokenApprovals;
```

Comment:
InflowNFT doesn't delete this but OutflowNFT does.

Reply:
good catch.